<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzes qPCR Data and Makes Plots for Visualization • WARPqpcr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Analyzes qPCR Data and Makes Plots for Visualization">
<meta property="og:description" content="Provides functionality for analyzing qPCR data using the dCT and ddCT methods and 
    generating plots of results. This package also contains functions for finding the most stable
    housekeeping gene from a group of possible genes.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">WARPqpcr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/shiny_app_usage.html">Using the WARPqpcr Web Application</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/sccallahan/WARPqpcr">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="warpqpcr" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#warpqpcr" class="anchor"></a>WARPqpcr</h1></div>
<!-- badges: start -->

<p><strong>W</strong>eb <strong>A</strong>pp and <strong>R</strong> <strong>P</strong>ackage for qPCR (WARPqpcr) provides functionality for analyzing qPCR data using the dCT and ddCT methods and generating plots of results. This package also contains functions for finding the most stable housekeeping gene from a group of possible genes.</p>
<p>The goal of this package is to allow rapid, reproducible analysis of qPCR data, particularly in the molecular biology setting, to provide easy functions for visualizing results, and to provide a tool for selecting the best housekeeping gene from a list of candidates. Many of the functions in this package rely on the ReadqPCR and NormqPCR packages for calculations, and the housekeeping gene stability is an implementation of an existing method (references for all of this are at the end of the README!)</p>
<div id="accessing-the-web-app" class="section level2">
<h2 class="hasAnchor">
<a href="#accessing-the-web-app" class="anchor"></a>Accessing the Web App</h2>
<p>In addition to being available as an R package, WARPqpcr is available as a web application. This web app requires no knowledge of R or programming in general to use. Users only need to upload their samplesheet (format described below), choose an analysis type, then fill out a few texts fields. The analysis is automatically run, and results (.txt files with CT values, ddCT values, etc. and .png files of relative expression) are available for download as a zip file.</p>
<p><a href="https://s-carson-callahan.shinyapps.io/WARPqpcr/">CLICK HERE</a> to go to the shinyapps.io page containing the web app.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the released version of WARPqpcr from <a href="https://github.com/sccallahan/WARPqpcr">this repo</a> with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install devtools if needed -- must have for installing from github</span>
<span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)

<span class="co"># now install WARPqpcr</span>
<span class="kw">install_github</span>(<span class="st">"sccallahan/WARPqpcr"</span>)</code></pre></div>
<p><strong>NB:</strong> Some linux systems may have difficulty installing some dependencies, particularly the <code>rgl</code> package (a dependency quite a way down the chain). This can be fixed by running the following lines in terminal:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">apt-get</span> install  libx11-dev mesa-common-dev libglu1-mesa-dev</code></pre></div>
</div>
<div id="some-notes-about-samplesheet-format-and-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#some-notes-about-samplesheet-format-and-methods" class="anchor"></a>Some notes about samplesheet format and methods</h2>
<div id="samplesheet-format-and-reading-in-data" class="section level4">
<h4 class="hasAnchor">
<a href="#samplesheet-format-and-reading-in-data" class="anchor"></a>Samplesheet format and reading in data</h4>
<p>The format for the samplesheet is described in the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ReadqPCR/inst/doc/ReadqPCR.pdf">ReadqPCR documentation.</a> In short, it requires 5 columns in a <em>tab-delimited</em> file:</p>
<ul>
<li>Well. The well letter/number.</li>
<li>Plate. The plate number.</li>
<li>Sample. The name of the sample being analayzed. It is <em>critical</em> that biological replicates have unique identifiers (e.g. “WT_1” and “WT_2” instead of just “WT” for both). Samples with identical names will be treated as technical replicates.</li>
<li>Detector. The gene being measured.</li>
<li>Cq. The Ct/Cq value for the well. These values <em>must</em> be either numerics or NAs.</li>
</ul>
<p>The <code>readSampleSheet</code> function assumes that the data is formatted as above and each sample is measuring the same genes with the same number of replicates per gene per sample (i.e. all genes measured must be in all samples, and each gene must have equal numbers of replicates). If you end up needing to discard replicates for certain genes/samples because they fail QC or the qPCR reaction failed, please leave the wells as just NAs instead of deleting the data. If your machine outputs non-NA values for wells where nothing was measured, please use the <code>readSampleSheet_NoCT</code> function to convert the data to NAs.</p>
<p>An example of the format required:</p>
<div class="sourceCode"><pre class="sourceCode R"><code class="sourceCode r"><span class="co">#&gt;    Well Plate Sample Detector    Cq</span>
<span class="co">#&gt; 1    A1     1     NT    GeneA 24.70</span>
<span class="co">#&gt; 2    A2     1    sh1    GeneA 24.98</span>
<span class="co">#&gt; 3    A3     1    sh3    GeneA 24.84</span>
<span class="co">#&gt; 4    A4     1    sh5    GeneA 24.45</span>
<span class="co">#&gt; 5    A5     1    sh6    GeneA 24.34</span>
<span class="co">#&gt; 6    A6     1     NT      HKG 16.79</span>
<span class="co">#&gt; 7    A7     1    sh1      HKG 16.78</span>
<span class="co">#&gt; 8    A8     1    sh3      HKG 17.14</span>
<span class="co">#&gt; 9    A9     1    sh5      HKG 17.32</span>
<span class="co">#&gt; 10  A10     1    sh6      HKG 16.97</span></code></pre></div>
</div>
<div id="methods" class="section level4">
<h4 class="hasAnchor">
<a href="#methods" class="anchor"></a>Methods</h4>
<p>The methods for biological replicates can be found in the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/NormqPCR/inst/doc/NormqPCR.pdf">NormqPCR documentation</a>. In short:</p>
<ul>
<li>The <code>singleRep</code> functions use the SD from the inital technical replicate merging to propagate error for downstream calculations.</li>
<li>The <code>bioRep</code> functions ignore technical replicate SD for error propagation - the <code>calcCV</code> function is provided to use the coefficient of variation as a filter for technical replication. The error propagation begins with calculating the mean and SD of biological replicates. The final ddCT subtraction is treated as subtraction from an arbitrary constant for error propagation.</li>
</ul>
</div>
</div>
<div id="example-for-single-biological-replicate-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-for-single-biological-replicate-data" class="anchor"></a>Example for single (biological) replicate data</h2>
<p>This mode is only recommended for pilot data or cases where several constructs (shRNA, overexpression, etc.) are being screened for efficiency. <em>No statistics are calculated</em> because there are no biological replicates. While statistics could be computed on this data, I do not feel comfortable recommending that approach, as it is simply measuring the user’s pipetting error.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(WARPqpcr))

<span class="co"># read in the qPCR data</span>
sampleObj &lt;-<span class="st"> </span><span class="kw"><a href="reference/readSampleSheet.html">readSampleSheet</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"samplesheet_singleRep.csv"</span>, <span class="dt">package =</span> <span class="st">"WARPqpcr"</span>))

<span class="co"># take a look at raw values</span>
rawct &lt;-<span class="st"> </span><span class="kw"><a href="reference/getRawCT.html">getRawCT</a></span>(<span class="dt">sampleObj =</span> sampleObj)

<span class="co"># calculate average values and coefficient of variation (CV)</span>
<span class="co"># This step is HIGHLY RECOMMENDED. The CV allows for a measure of technical replicate consistency </span>
avgCT &lt;-<span class="st"> </span><span class="kw"><a href="reference/getAvgCT.html">getAvgCT</a></span>(<span class="dt">sampleObj =</span> sampleObj)
avgCT &lt;-<span class="st"> </span><span class="kw"><a href="reference/calcCV.html">calcCV</a></span>(<span class="dt">avgCtObj =</span> avgCT)
<span class="co">#&gt; All samples passed the CV threshold!</span>

<span class="co"># if we want, we can make a plot of these avgCT values</span>
<span class="kw"><a href="reference/plotAvgCT.html">plotAvgCT</a></span>(<span class="dt">avgCtObj =</span> avgCT, <span class="dt">theme_classic =</span> <span class="ot">TRUE</span>, <span class="dt">title =</span> <span class="st">"Avg CT"</span>)</code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># calculate dCT values and plot them</span>
<span class="co"># we can assign our HKG to a vector or pass it as an argument directly</span>
hkg &lt;-<span class="st"> "HKG"</span>
dct &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_dCT_singleRep.html">get_dCT_singleRep</a></span>(<span class="dt">sampleObj =</span> sampleObj, <span class="dt">hkg =</span> hkg)
<span class="kw"><a href="reference/plot_dCT_singleRep.html">plot_dCT_singleRep</a></span>(dct, <span class="dt">theme_classic =</span> <span class="ot">TRUE</span>, <span class="dt">title =</span> <span class="st">"dCT values"</span>)</code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-2.png" width="100%"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Now we can calculate ddCT values and plot -- this is typically the end result desired</span>
<span class="co"># We can technically skip straight to this step, but calculating the CV first is recommended</span>
<span class="co"># we need to indicate which sample is the control</span>
control &lt;-<span class="st"> "NT"</span>
ddct &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_ddCT_singleRep.html">get_ddCT_singleRep</a></span>(<span class="dt">sampleObj =</span> sampleObj, <span class="dt">rel.exp =</span> <span class="ot">TRUE</span>, <span class="dt">hkg =</span> hkg, <span class="dt">control =</span> control) <span class="co"># same hkg as the dCT step above</span>
<span class="kw"><a href="reference/plot_ddCT_singleRep.html">plot_ddCT_singleRep</a></span>(ddct, <span class="dt">theme_classic =</span> <span class="ot">TRUE</span>, <span class="dt">rel.exp =</span> <span class="ot">TRUE</span>, <span class="dt">title =</span> <span class="st">"Relative Expression (ddCT)"</span>)</code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-3.png" width="100%"></p>
</div>
<div id="example-for-biological-replicate-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-for-biological-replicate-data" class="anchor"></a>Example for biological replicate data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(WARPqpcr))

<span class="co"># read in the qPCR data</span>
sampleObj &lt;-<span class="st"> </span><span class="kw"><a href="reference/readSampleSheet.html">readSampleSheet</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"samplesheet_bioReps.csv"</span>, <span class="dt">package =</span> <span class="st">"WARPqpcr"</span>))

<span class="co"># check raw values</span>
rawct &lt;-<span class="st"> </span><span class="kw"><a href="reference/getRawCT.html">getRawCT</a></span>(<span class="dt">sampleObj =</span> sampleObj)

<span class="co"># merge technical replicates and plot</span>
avgct &lt;-<span class="st"> </span><span class="kw"><a href="reference/getAvgCT.html">getAvgCT</a></span>(<span class="dt">sampleObj =</span> sampleObj)
avgct &lt;-<span class="st"> </span><span class="kw"><a href="reference/calcCV.html">calcCV</a></span>(avgct)
<span class="co">#&gt; 3 samples have CV over the threshold, there may be issues with technical replication</span>
<span class="co">#&gt; Check CV column for affected samples and raw data for outlying replicates</span>
<span class="co"># samples have higher CVs; for the sake of example we will continue, but in real data this should be investigated</span>
<span class="kw"><a href="reference/plotAvgCT.html">plotAvgCT</a></span>(avgct, <span class="dt">title =</span> <span class="st">"Avg CT values"</span>)</code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-1.png" width="100%"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># set our housekeeping gene and calculate dCT</span>
hkg &lt;-<span class="st"> "HKG"</span>
dct &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_dCT_bioReps.html">get_dCT_bioReps</a></span>(<span class="dt">sampleObj =</span> sampleObj, <span class="dt">hkg =</span> hkg)

<span class="co"># since we have biological replicates, we can calculate significance</span>
<span class="co"># stats should be performed on these dCT values</span>
signifTest &lt;-<span class="st"> </span><span class="kw"><a href="reference/signifTest.html">signifTest</a></span>(<span class="dt">dCTObj =</span> dct, <span class="dt">gene.name =</span> <span class="st">"GeneA"</span>, <span class="dt">var.equal =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(signifTest)
<span class="co">#&gt;            pvals</span>
<span class="co">#&gt; GeneA 0.02594038</span>

<span class="co"># now we can calculate the ddCT to get expression changes and make plots</span>
<span class="co"># we will do both relative expression and log2 fold-change for the sake of demonstration</span>

<span class="co"># RELATIVE EXPRESSION</span>
relExp &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_ddCT_bioReps.html">get_ddCT_bioReps</a></span>(<span class="dt">sampleObj =</span> sampleObj, <span class="dt">cond.1 =</span> <span class="st">"KD"</span>, <span class="dt">cond.2 =</span> <span class="st">"LUC"</span>, <span class="dt">reps.cond.1 =</span> <span class="dv">3</span>, <span class="dt">reps.cond.2 =</span> <span class="dv">3</span>,
                 <span class="dt">hkg =</span> hkg, <span class="dt">rel.exp =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="reference/plot_ddCT_bioReps.html">plot_ddCT_bioReps</a></span>(<span class="dt">ddCTobj =</span> relExp, <span class="dt">theme_classic =</span> <span class="ot">TRUE</span>,
                  <span class="dt">rel.exp =</span> <span class="ot">TRUE</span>, <span class="dt">title =</span> <span class="st">"Relative Expression (ddCT)"</span>) <span class="co">#NB: rel.exp value must match data type</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-2.png" width="100%"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># LOG2 FOLD-CHANGE</span>
absExp &lt;-<span class="st"> </span><span class="kw"><a href="reference/get_ddCT_bioReps.html">get_ddCT_bioReps</a></span>(<span class="dt">sampleObj =</span> sampleObj, <span class="dt">cond.1 =</span> <span class="st">"KD"</span>, <span class="dt">cond.2 =</span> <span class="st">"LUC"</span>, <span class="dt">reps.cond.1 =</span> <span class="dv">3</span>, <span class="dt">reps.cond.2 =</span> <span class="dv">3</span>,
                 <span class="dt">hkg =</span> hkg, <span class="dt">rel.exp =</span> <span class="ot">FALSE</span>)
<span class="kw"><a href="reference/plot_ddCT_bioReps.html">plot_ddCT_bioReps</a></span>(<span class="dt">ddCTobj =</span> absExp, <span class="dt">theme_classic =</span> <span class="ot">TRUE</span>,
                  <span class="dt">rel.exp =</span> <span class="ot">FALSE</span>, <span class="dt">title =</span> <span class="st">"log2 fold-change (ddCT)"</span>) <span class="co">#NB: rel.exp value must match data type</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-3.png" width="100%"></p>
</div>
<div id="example-for-selecting-most-stable-housekeeping-gene" class="section level2">
<h2 class="hasAnchor">
<a href="#example-for-selecting-most-stable-housekeeping-gene" class="anchor"></a>Example for selecting most stable housekeeping gene</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(WARPqpcr))

<span class="co"># read in the qPCR data</span>
sampleObj &lt;-<span class="st"> </span><span class="kw"><a href="reference/readSampleSheet.html">readSampleSheet</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"samplesheet_hkgStab.csv"</span>, <span class="dt">package =</span> <span class="st">"WARPqpcr"</span>))

<span class="co"># get average CT values</span>
avgCT &lt;-<span class="st"> </span><span class="kw"><a href="reference/getAvgCT.html">getAvgCT</a></span>(<span class="dt">sampleObj =</span> sampleObj)

<span class="co"># calculate stability values</span>
hkgs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gene1"</span>, <span class="st">"gene2"</span>, <span class="st">"gene3"</span>, <span class="st">"gene4"</span>)
hkgStab &lt;-<span class="st"> </span><span class="kw"><a href="reference/calcStab.html">calcStab</a></span>(<span class="dt">avgCTobj =</span> avgCT, <span class="dt">hkgs =</span> hkgs)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(hkgStab)
<span class="co">#&gt;              V1</span>
<span class="co">#&gt; gene4 0.4562171</span>
<span class="co">#&gt; gene3 0.4562171</span>
<span class="co">#&gt; gene2 0.5965916</span>
<span class="co">#&gt; gene1 0.9475278</span></code></pre></div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ul>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-296">ReadqPCR and NormqPCR</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/17026756">Selection of housekeeping genes for gene expression studies in human reticulocytes using real-time PCR</a></li>
</ul>
</div>
<div id="to-do" class="section level2">
<h2 class="hasAnchor">
<a href="#to-do" class="anchor"></a>To do</h2>
<ul>
<li>[ ] Port to Shiny App</li>
<li>[ ] Add vignettes/links for web app to repo</li>
<li>[ ] ChIP-qPCR analysis module</li>
<li>[ ] Add examples to README</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/sccallahan/WARPqpcr">https://​github.com/​sccallahan/​WARPqpcr</a>
</li>
<li>Report a bug at <br><a href="https://github.com/sccallahan/WARPqpcr/issues">https://​github.com/​sccallahan/​WARPqpcr/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>S. Carson Callahan <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/sccallahan/WARPqpcr"><img src="https://travis-ci.org/sccallahan/WARPqpcr.svg?branch=master" alt="Build Status"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by S. Carson Callahan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
